workflow/util/util.go

```diff
@@ -805,7 +805,9 @@ func FormulateRetryWorkflow(ctx context.Context, wf *wfv1.Workflow, restartSucce
 	switch wf.Status.Phase {
 	case wfv1.WorkflowFailed, wfv1.WorkflowError:
 	default:
-		return nil, nil, errors.Errorf(errors.CodeBadRequest, "workflow must be Failed/Error to retry")
+		if !(restartSuccessful && len(nodeFieldSelector) > 0) {
+			return nil, nil, errors.Errorf(errors.CodeBadRequest, "To retry either: (1) workflow must be Failed/Error (2) Provide also restartSuccessful and nodeFieldSelector value")
```

Comment:
This might be a bit hard to understand.

Reply:
I know. Do you have an idea on how to simplify it? Maybe a link to docs?

Reply:
Looks good

Reply:
```suggestion
			return nil, nil, errors.Errorf(errors.CodeBadRequest, "To retry a successful workflow, set the options restartSuccessful and nodeFieldSelector")
```

Reply:
done